server/api/user/user.model.js

```diff
@@ -90,90 +108,409 @@ function isoDateToTime(isoDate){
+                    foundYear = true;
+                    var thisYear = this.attendanceByYear[i];
+
+                    if (status === 'present' || status === 'presentSmall' || status === 'presentBonus'){
+
+                        // Check if user is unverified for today
+                        for (var a = thisYear.unverified.length-1; a>= 0 ; a--){
+                          if (isoDateToTime(thisYear.unverified[a].date) === isoDateToTime(today)){
+
+                            var thisDate = thisYear.unverified[a];
+                            var found = false;
+                            if (status === 'present' && thisDate.bonus === false && thisDate.smallgroup === false){
+                                found = true;
+                            }
+                            if (status === 'presentSmall' && thisDate.bonus === false && thisDate.smallgroup === true){
+                                found = true;
+                            }
+                            if (status === 'presentBonus' && thisDate.bonus === true && thisDate.smallgroup === false){
+                                found = true;
+                            }
+                            if (found){
+                                thisYear.unverified[a].splice(a,1);
+                                thisYear.verified.push(thisDate);
+                                this.attendanceByYear[i]=thisYear;
+                                // this.attendanceByYear[i].verified.set()
+                                this.markModified("attendanceByYear");
+                                thisYear.save();
+                                return;
+                            }
+                          }
+                        }
+
+                        // Check if user is already verified for today
+                        for (var a = thisYear.verified.length-1; a>= 0 ; a--){
+
+                          if (isoDateToTime(thisYear.verified[a].date) === isoDateToTime(today)){
+
+                            var thisDate = thisYear.verified[a];
+                            var found = false;
+                            if (status === 'present' && thisDate.bonus === false && thisDate.smallgroup === false){
+                                found = true;
+                            }
+                            if (status === 'presentSmall' && thisDate.bonus === false && thisDate.smallgroup === true){
+                                found = true;
+                            }
+                            if (status === 'presentBonus' && thisDate.bonus === true && thisDate.smallgroup === false){
+                                found = true;
+                            }
+                            if (found){
+                                return;
+                            }
+                          }
+
+                        }
+                        if (status === 'present'){
```

Comment:
I was trying to leave flexibility in there.  As I worked in different parts, there are additional functions / properties that could be changed, but this was the original scheme.  It is intended to allow codes of all times to be generated on any one calendar day.